### PR TITLE
update vision/nlp sdkv2 e2e tests to use components

### PR DESCRIPTION
**Pull request by bhimar**
# Description

Modify the e2e tests for vision and nlp to use the new components and models

Runs:

Image Multiclass:
[ft component sweep](https://ml.azure.com/experiments/id/63ba9bf3-a3ea-46ce-a0da-13f1ec3dd36d/runs/musing_brain_nbf514lgxv_HD?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rbhimani-rg/providers/Microsoft.MachineLearningServices/workspaces/rbhimani-ws&tid=72f988bf-86f1-41af-91ab-2d7cd011db47)
[runtime component sweep](https://ml.azure.com/runs/careful_nose_9bt5wrxc48?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rbhimani-rg/providers/Microsoft.MachineLearningServices/workspaces/rbhimani-ws&tid=72f988bf-86f1-41af-91ab-2d7cd011db47)
[automode runtime component](https://ml.azure.com/experiments/id/63ba9bf3-a3ea-46ce-a0da-13f1ec3dd36d/runs/purple_candle_568fk85w2l?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rbhimani-rg/providers/Microsoft.MachineLearningServices/workspaces/rbhimani-ws&tid=72f988bf-86f1-41af-91ab-2d7cd011db47)

Image Multilabel:
[ft component sweep](https://ml.azure.com/experiments/id/63ba9bf3-a3ea-46ce-a0da-13f1ec3dd36d/runs/gentle_watch_4kzf6b5bf5?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rbhimani-rg/providers/Microsoft.MachineLearningServices/workspaces/rbhimani-ws&tid=72f988bf-86f1-41af-91ab-2d7cd011db47) --> Note the parent run fails due to a primary metric not being reported, this bug is in PR for JOS repo
[runtime component  sweep](https://ml.azure.com/experiments/id/63ba9bf3-a3ea-46ce-a0da-13f1ec3dd36d/runs/quiet_nerve_v60qcsj7yg?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rbhimani-rg/providers/Microsoft.MachineLearningServices/workspaces/rbhimani-ws&tid=72f988bf-86f1-41af-91ab-2d7cd011db47)
[automode runtime component](https://ml.azure.com/experiments/id/63ba9bf3-a3ea-46ce-a0da-13f1ec3dd36d/runs/purple_candle_568fk85w2l?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rbhimani-rg/providers/Microsoft.MachineLearningServices/workspaces/rbhimani-ws&tid=72f988bf-86f1-41af-91ab-2d7cd011db47)

Object Detection:
[ft_component_sweep](https://ml.azure.com/experiments/id/63ba9bf3-a3ea-46ce-a0da-13f1ec3dd36d/runs/eager_monkey_sxzkjms58n?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rbhimani-rg/providers/Microsoft.MachineLearningServices/workspaces/rbhimani-ws&tid=72f988bf-86f1-41af-91ab-2d7cd011db47)
[runtime component sweep](https://ml.azure.com/runs/tidy_sun_n1c5qvpmj7?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rbhimani-rg/providers/Microsoft.MachineLearningServices/workspaces/rbhimani-ws&tid=72f988bf-86f1-41af-91ab-2d7cd011db47)
[automode runtime component](https://ml.azure.com/experiments/id/63ba9bf3-a3ea-46ce-a0da-13f1ec3dd36d/runs/tough_zebra_zhcmczltjz?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rbhimani-rg/providers/Microsoft.MachineLearningServices/workspaces/rbhimani-ws&tid=72f988bf-86f1-41af-91ab-2d7cd011db47)

Instance Segmentation:
[ft_component_sweep](https://ml.azure.com/experiments/id/63ba9bf3-a3ea-46ce-a0da-13f1ec3dd36d/runs/frosty_kettle_ddsss3yyv6?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rbhimani-rg/providers/Microsoft.MachineLearningServices/workspaces/rbhimani-ws&tid=72f988bf-86f1-41af-91ab-2d7cd011db47)
[runtime component sweep](https://ml.azure.com/experiments/id/63ba9bf3-a3ea-46ce-a0da-13f1ec3dd36d/runs/gray_tongue_vzywp8lx4y?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rbhimani-rg/providers/Microsoft.MachineLearningServices/workspaces/rbhimani-ws&tid=72f988bf-86f1-41af-91ab-2d7cd011db47)
[automode runtime component](https://ml.azure.com/experiments/id/63ba9bf3-a3ea-46ce-a0da-13f1ec3dd36d/runs/tough_flag_rw363v3rms?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rbhimani-rg/providers/Microsoft.MachineLearningServices/workspaces/rbhimani-ws&tid=72f988bf-86f1-41af-91ab-2d7cd011db47)

NLP:
[classification](https://ml.azure.com/experiments/id/025b19a9-783c-49f8-8156-328fc4967542/runs/affable_stamp_dr707vctgw?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rbhimani-rg/providers/Microsoft.MachineLearningServices/workspaces/rbhimani-ws&tid=72f988bf-86f1-41af-91ab-2d7cd011db47)
[multilabel](https://ml.azure.com/runs/plum_glass_d9smvc2y7n?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rbhimani-rg/providers/Microsoft.MachineLearningServices/workspaces/rbhimani-ws&tid=72f988bf-86f1-41af-91ab-2d7cd011db47)
[ner](https://ml.azure.com/runs/teal_melon_8yx7gp39yl?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rbhimani-rg/providers/Microsoft.MachineLearningServices/workspaces/rbhimani-ws&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#)

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by bhimar** [comment]
TODO: change NLP component id's to registry accessbile in eastus

**Review by MaurisLucis — APPROVED** [review]
Approving, but let's please change the properties for the nlp test to use registry components rather than pointing to some random dev workspace as soon as we can. thanks!

**Comment by bhimar** [comment]
> Approving, but let's please change the properties for the nlp test to use registry components rather than pointing to some random dev workspace as soon as we can. thanks!

@MaurisLucis Will do this before merging the PR. WIP uploading and testing nlp registry components

File: sdk/ml/azure-ai-ml/tests/automl_job/e2etests/test_automl_image_classification.py [hunk 1]
```diff
@@ -117,6 +119,24 @@ def test_image_classification_multiclass_run(
...
 
+        # Configure component sweep job
+        image_classification_job_component_sweep = copy.deepcopy(image_classification_job)
+        image_classification_job_component_sweep.set_training_parameters(early_stopping=True, evaluation_frequency=1)
+        image_classification_job_component_sweep.extend_search_space(
+            [
+                SearchSpace(
+                    model_name=Choice(["microsoft/beit-base-patch16-224"]),
+                    learning_rate=Uniform(0.001, 0.01),
+                    number_of_epochs=Choice([15, 30]),
```
Comment (raduk) [inline sdk/ml/azure-ai-ml/tests/automl_job/e2etests/test_automl_image_classification.py:130]: any reason to use more than one epoch? number_of_epochs=Choice([1])

we need to run tests as fast as possible

File: sdk/ml/azure-ai-ml/tests/automl_job/e2etests/test_automl_image_classification.py [hunk 2]
```diff
@@ -83,6 +83,8 @@ def test_image_classification_multiclass_run(
 
         training_data = Input(type=AssetTypes.MLTABLE, path=train_path)
         validation_data = Input(type=AssetTypes.MLTABLE, path=val_path)
+        properties = get_automl_job_properties()
+        properties['_pipeline_id_override'] = "azureml://registries/azmlft-dev-registry01/components/image_classification_pipeline/versions/0.0.5"
```
Comment (raduk) [inline sdk/ml/azure-ai-ml/tests/automl_job/e2etests/test_automl_image_classification.py:87]: any way we can use the latest version instead of a specific one? also, why do we need to specify it?

File: sdk/ml/azure-ai-ml/tests/automl_job/e2etests/test_automl_image_classification_multilabel.py [hunk 3]
```diff
@@ -128,6 +130,27 @@ def test_image_classification_multilabel_run(
...
+        image_classification_multilabel_job_component_sweep = copy.deepcopy(image_classification_multilabel_job)
+        image_classification_multilabel_job_component_sweep.set_training_parameters(
+            early_stopping=True, evaluation_frequency=1
+        )
+        image_classification_multilabel_job_component_sweep.extend_search_space(
+            [
+                SearchSpace(
+                    model_name=Choice(["microsoft/beit-base-patch16-224"]),
+                    learning_rate=Uniform(0.005, 0.05),
+                    number_of_epochs=Choice([15, 30]),
```
Comment (raduk) [inline sdk/ml/azure-ai-ml/tests/automl_job/e2etests/test_automl_image_classification_multilabel.py:143]: same comment as above

File: sdk/ml/azure-ai-ml/tests/automl_job/e2etests/test_automl_image_object_detection.py [hunk 4]
```diff
@@ -152,6 +155,23 @@ def test_image_object_detection_run(
             early_termination=BanditPolicy(evaluation_interval=2, slack_factor=0.2, delay_evaluation=6),
         )
 
+        # Configure component sweep job
+        image_object_detection_job_component_sweep = copy.deepcopy(image_object_detection_job)
+        image_object_detection_job_component_sweep.set_training_parameters(early_stopping=True, evaluation_frequency=1)
```
Comment (raduk) [inline sdk/ml/azure-ai-ml/tests/automl_job/e2etests/test_automl_image_object_detection.py:160]: make sure it is only one epoch?

File: sdk/ml/azure-ai-ml/tests/automl_job/e2etests/test_automl_image_segmentation.py [hunk 5]
```diff
@@ -105,6 +108,26 @@ def test_image_segmentation_run(self, image_segmentation_dataset: Tuple[Input, I
             early_termination=BanditPolicy(evaluation_interval=2, slack_factor=0.2, delay_evaluation=6),
         )
 
+        # Configure component sweep job
+        image_instance_segmentation_job_component_sweep = copy.deepcopy(image_instance_segmentation_job)
+        image_instance_segmentation_job_component_sweep.set_training_parameters(
```
Comment (raduk) [inline sdk/ml/azure-ai-ml/tests/automl_job/e2etests/test_automl_image_segmentation.py:113]: make sure it is only one epoch?

File: sdk/ml/azure-ai-ml/tests/automl_job/e2etests/test_automl_image_segmentation.py [hunk 6]
```diff
@@ -75,6 +75,9 @@ def test_image_segmentation_run(self, image_segmentation_dataset: Tuple[Input, I
         training_data = Input(type=AssetTypes.MLTABLE, path=train_path)
         validation_data = Input(type=AssetTypes.MLTABLE, path=val_path)
 
+        properties = get_automl_job_properties()
```
Comment (raduk) [inline sdk/ml/azure-ai-ml/tests/automl_job/e2etests/test_automl_image_segmentation.py:78]: by doing this, aren't we only launching jobs with the components?

File: sdk/ml/azure-ai-ml/tests/automl_job/e2etests/test_automl_image_classification.py (hunk 2 above)
Comment (bhimar) [reply sdk/ml/azure-ai-ml/tests/automl_job/e2etests/test_automl_image_classification.py:87]: We are specifying the pipeline component id because JOS currently points to azureml-dev registry which is not accessible outside of centraluseuap. However, the nighly sdk tests run in eastus, so we need to have components in a registry that is accessible in prod regions.

Comment (bhimar) [reply sdk/ml/azure-ai-ml/tests/automl_job/e2etests/test_automl_image_classification.py:87]: Will check on latest version, it should be possible

File: sdk/ml/azure-ai-ml/tests/automl_job/e2etests/test_automl_image_segmentation.py (hunk 6 above)
Comment (bhimar) [reply sdk/ml/azure-ai-ml/tests/automl_job/e2etests/test_automl_image_segmentation.py:78]: That's correct, we are only launching jobs with components. Would it be better to keep the runtime tests and duplicate for components even if the runtime component is tested?

Comment (raduk) [reply sdk/ml/azure-ai-ml/tests/automl_job/e2etests/test_automl_image_segmentation.py:78]: but this PR is removing the tests for the current e2e workflow, no? we want to add component tests on top of existing ones, not to replace them ...

Comment (bhimar) [reply sdk/ml/azure-ai-ml/tests/automl_job/e2etests/test_automl_image_segmentation.py:144]: What are your thoughts on the [dsl tests](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_automl_dsl_pipeline.py)? The run is kicked off and immediately cancelled for the rest object can be inspected, so the behavior is the same with or without the feature flag.

**Review by PhaniShekhar — APPROVED** [review]

File: sdk/ml/azure-ai-ml/tests/automl_job/e2etests/test_automl_image_segmentation.py (hunk 6 above)
Comment (bhimar) [reply sdk/ml/azure-ai-ml/tests/automl_job/e2etests/test_automl_image_segmentation.py:144]: Discussed offline with Phani, we decided that because dsl tests don't actually submit runs that we shouldn't duplicate them.

File: sdk/ml/azure-ai-ml/tests/automl_job/e2etests/test_automl_image_classification.py [hunk 7]
```diff
@@ -136,3 +136,102 @@ def test_image_classification_multiclass_run(
...
+            sampling_algorithm="Random",
+            early_termination=BanditPolicy(evaluation_interval=2, slack_factor=0.2, delay_evaluation=6),
+        )
+
+        # Configure AutoMode job
+        image_classification_job_automode = copy.deepcopy(image_classification_job)
+        # TODO: after shipping the AutoMode feature, do not set flag and call `set_limits()` instead of changing
+        # the limits object directly.
+        image_classification_job_automode.properties["enable_automode"] = True
+        image_classification_job_automode.limits.max_trials = 2
```
Comment (raduk) [inline sdk/ml/azure-ai-ml/tests/automl_job/e2etests/test_automl_image_classification.py:218]: why 2?

**Review by raduk — APPROVED** [review]

File: sdk/ml/azure-ai-ml/tests/automl_job/e2etests/test_automl_image_classification.py (hunk 7 above)
Comment (bhimar) [reply sdk/ml/azure-ai-ml/tests/automl_job/e2etests/test_automl_image_classification.py:134]: resolved offline

**Review by luigiw — APPROVED** [review]

**Review by raduk — APPROVED** [review]
Great to see the refactoring! We do have a lot of duplications across different tasks (before your PR), but that's for another time :)
